extensions/amp-analytics/0.1/config.js

```diff
@@ -270,6 +271,47 @@ export class AnalyticsConfig {
     });
   }
 
+  /**
+   * Check to see if config has warning, display on console and
+   * remove the key.
+   * @private
+   * @return {!Promise<undefined>}
+   */
+  checkWarningMessage_() {
+    if (this.config_['warningMessage']) {
+      const TAG = this.getName_();
+      const formattedMessage = this.getFormattedWarningMessage_();
+      user().warn(TAG, formattedMessage);
+      delete this.config_['warningMessage'];
+    }
+    return Promise.resolve();
+  }
+
+  /**
+   * Build the formatted warning message that
+   * displays the vendor type and/or url.
+   * Eg: Warning from ./analytics.config.json: I am deprecated
+   * @private
+   * @return {string}
+   */
+  getFormattedWarningMessage_() {
+    const type = this.element_.getAttribute('type');
+
+    let configUrl = type
+      ? this.getVendorUrl_(type)
```

Comment:
`getVendorUrl_` is used to calculate the remote vendor config location. The `configUrl` should be null if the `config` attribute is not specified.